### PR TITLE
Kozyreva Ekaterina. Lab 1. Option 1

### PR DESCRIPTION
**Pull request by kth-vyu**

### CODE REVIEW

File: clang/test/lab1/kozyreva_ekaterina/tests.cpp [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+// CHECK-NEXT: |_var
+
+class Test4 {
+  class Nested {
+    int z;
+  };
+};
+// CHECK: Test4
+// CHECK: Nested
+// CHECK-NEXT: |_z
```
Comment (m-ly4) [inline clang/test/lab1/kozyreva_ekaterina/tests.cpp]: Please add a test for static member fields also.

File: clang/labs/lab1/kozyreva_ekaterina/PrintNames.cpp [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+public:
+  std::unique_ptr<clang::ASTConsumer>
+  CreateASTConsumer(clang::CompilerInstance &ci, llvm::StringRef) override {
+    return std::make_unique<ClassPrinterASTConsumer>();
+  }
+
+  bool ParseArgs(const clang::CompilerInstance &ci,
+                 const std::vector<std::string> &Args) override {
+    for (const auto &arg : Args) {
+      if (arg == "--help") {
```
Comment (m-ly4) [inline clang/labs/lab1/kozyreva_ekaterina/PrintNames.cpp]: Why `true` is returned if `help` argument is specified?

**Review by m-ly4 — CHANGES_REQUESTED** [review]

File: clang/labs/lab1/kozyreva_ekaterina/PrintNames.cpp (hunk 2 above)
Comment (kth-vyu) [reply clang/labs/lab1/kozyreva_ekaterina/PrintNames.cpp]: Thanks, I fixed it, it should return false and stop the program.

File: clang/test/lab1/kozyreva_ekaterina/tests.cpp (hunk 1 above)
Comment (m-ly4) [reply clang/test/lab1/kozyreva_ekaterina/tests.cpp]: Still static members printing is required.

Comment (kth-vyu) [reply clang/test/lab1/kozyreva_ekaterina/tests.cpp]: I'm sorry, I've tried everything. When I add a test for static fields, it doesn't work. Tell me, is there a bug in the plugin somewhere or am I writing tests wrong somehow? I can't think of anything anymore, I haven't found anything.

Comment (m-ly4) [reply clang/test/lab1/kozyreva_ekaterina/tests.cpp]: Static fields of a class are saved in a bit different nodes of AST tree. That is why your plugin doesn't print them now. You may ask your groupmates who already made your variant regarding printing static fields.

Comment (aobolensk) [reply clang/test/lab1/kozyreva_ekaterina/tests.cpp]: Static fields comment is not resolved @E-Kozyreva

**Review by m-ly4 — APPROVED** [review]

**Review by aobolensk — APPROVED** [review]

**Pull request merged**